packages/ui/src/PhoneNumberField.tsx

```diff
@@ -0,0 +1,92 @@
+import {AsYouType, parsePhoneNumberFromString} from "libphonenumber-js";
+import React, {FC, useCallback, useState} from "react";
+
+import {PhoneNumberFieldProps} from "./Common";
+import {TextField} from "./TextField";
+
+export const PhoneNumberField: FC<PhoneNumberFieldProps> = ({
+  errorText,
+  iconName,
+  placeholderText,
+  value,
+  onChange,
+  onBlur,
+  defaultCountryCode = "US",
+  ...rest
+}) => {
+  const [localValue, setLocalValue] = useState<string>(value || "");
+  const [error, setError] = useState<string | undefined>(errorText);
+
+  const validatePhoneNumber = useCallback(
+    (phoneNumber: string): string | undefined => {
+      if (phoneNumber.trim() === "") {
+        return undefined;
+      }
+      const parsedNumber = parsePhoneNumberFromString(phoneNumber, defaultCountryCode);
+      if (!parsedNumber) {
+        return "Invalid phone number format";
+      }
+      if (!parsedNumber.isPossible()) {
+        return "Phone number is not possible";
+      }
+      if (!parsedNumber.isValid()) {
+        return "Phone number is not valid";
+      }
+      return undefined;
+    },
+    [defaultCountryCode]
+  );
+
+  const formatPhoneNumber = useCallback(
+    (phoneNumber: string): string => {
+      const asYouType = new AsYouType(defaultCountryCode);
+      return asYouType.input(phoneNumber);
+    },
+    [defaultCountryCode]
+  );
+
+  const handleBlur = useCallback(
+    (inputValue: string) => {
+      if (onBlur) {
+        onBlur(inputValue);
+      }
+      const validationError = validatePhoneNumber(inputValue);
+      if (validationError) {
+        setError(validationError);
+      } else {
+        setError(undefined);
+      }
+    },
+    [onBlur, validatePhoneNumber]
+  );
+
+  const handleChange = useCallback(
```

Comment:
If you try to backspace past the `)` once you already have a valid phone number, it doesn't work unless you arrow over to the numbers and past the parenthesis.